drafts/2019-03-15-integration-testing-with-testcontainers.adoc

```diff
@@ -0,0 +1,317 @@
+---
+layout: post
+title:  "Easy integration testing for JavaEE and MicroProfile applications with Testcontainers"
+date:   2019-03-16 07:45:00 -0000
+categories: blog
+author_picture: https://avatars2.githubusercontent.com/u/5427967
+author_github: https://github.com/aguibert
+seo-title: Easy integration testing for JavaEE and MicroProfile applications with Testcontainers
+seo-description: Learn how to write simple and effective integration tests that run against your real application's docker container. No mock objects needed. 
+blog_description: "Learn how to write simple and effective integration tests that run against your real application's docker container. No mock objects needed."
+---
+:description: Learn how to write simple and effective integration tests that test your real application as it would run in production
+=  Easy integration testing for JavaEE and MicroProfile applications with Testcontainers
+Andy Guibert <https://github.com/aguibert>
+
+Have you ever toiled with creating mock objects for unit tests? How about custom setup steps for integration tests? Ever had an issue in production because of differences in behavior between testing and production environments?
+
+If you answered yes to any of these questions, answer one more for me: do your applications run in docker containers?
+
+One of the great benefits of docker is that we get a nice consistent package that contains everything down to the OS, meaning it's portable to any hardware. Great, so lets leverage this to get consistent testing environments too!
+
+== Introducing Testcontainers
+
+image::https://d33wubrfki0l68.cloudfront.net/a661dbbe55be3e9cb77889f24835a44c6daf53c2/ce0aa/logo.png[Testcontainers logo, align="center"]
+
+https://www.testcontainers.org/[Testcontainers] is a great open source library that essentially wraps Docker in a Java API, and has first-class integration with JUnit. It is mainly used for external resource dependencies, and has first-class support for many popular databases such as Postgres and MySQL. In addition, it also has generic support for any docker container.
+
```

Comment:
We need more of a "wow!" statement here.  Something to catch their attention.  The "in addition, it has generic support for docker.." just sounds like a "oh yeah, it has this too".  Just kind of a ho-hum statement, but this is really what you are selling this blog post.  Somehow, we need to highlight this fact before jumping into the Gradle dependencies.  Maybe a section subtitle of "Testcontainers and Application Servers"  or "Liberty and Testcontainers".  Something to show at a quick glance that this is the gist of the article.  The point of the blog just didn't jump out at me.

Reply:
Plus what he says ^^^ :)